### DISCUSSION TITLE
Chinese Special Features

### DISCUSSION THREAD

**Discussion by BernieHuang2008** [searxng/searxng#2803] [Ideas]
Many things in China/Chinese is different from the world. This discussion is designed to report all the Chinese-special-featutes and the solution to them.

**Comment by BernieHuang2008** [comment]
For example, the Chinese government has blocked some websites. Including `Twitter`, `wiki`, and even `google` and `github (depends on luck)`.

The Chinese can only access these websites with VPN. Unfortunately, VPN is not allowed in China (according to the goverment). 

So if the search result contains these blocked-websites, most of the users cannot access it!

**Reply by BernieHuang2008** [reply]
I'm planning to build a list which contains all the blocked-results, and hide them when a `chinese-plugin` is on or something.

**Reply by return42** [reply]
> Chinese government has blocked some websites. Including Twitter

What about alternatives like [nitter](https://nitter.net/search?f=users&q=foo)?

**Reply by BernieHuang2008** [reply]
It's not so simple as you think --- we searched for years, only some small alternatives are alive, and they are used as personal alternatives.

**Reply by return42** [reply]
OK I understand / sorry I'm a little bit naive :)

**Comment by extremelyonline** [comment]
Hi, I'm interested in this issue. Would you please test if the following frontend are blocked in China?

Twitter: nitter.d420.de
Wiki: wikiless.northboot.xyz
GitHub: g.opnxng.com

While Google is blocked, you should be able to just get Google results with Searxng, right?

I'll try my best to implement these frontends on my instance opnxng.com if they are not blocked!

**Reply by BernieHuang2008** [reply]
Thank you for your support!

Sorry, but the first two are completely blocked, while the third one is a little slow(may be the server is slow). 

And, yes, I can only get google result via searxng in normal ways.

I think there's just a very little mirror sites available in China.

My plan is just tag some results which is unavailable.

However, if you found any available mirror site, I'm glad to change my plan!

Thank you for your reply, again.

**Comment by extremelyonline** [comment]
If you host your own instance, you can simply hide domains using hostname_replace:

```
  # block bload domains
  '(.*\.)?github\.com': false
```

**Reply by BernieHuang2008** [reply]
Thanks for this advice very much.
I shall leave it here in case some host wants to use your way.

But what I'm plan to do is develop a way that can serve both Chinese & Non-chinese user.
